### PR TITLE
[sailfish-browser] Make overlay to open to top (y: 0) if no tabs exists. Fixes JB#53170

### PR DESCRIPTION
**Pull request by aroditelev**
https://bz.jollamobile.com/show_bug.cgi?id=53170

### CODE REVIEW

File: apps/shared/OverlayAnimator.qml [hunk 1]
```diff
@@ -162,6 +167,22 @@ Item {
             ]
         },
 
+        State {
+            name: _startPage
+            changes: [
+                PropertyChanges {
+                    target: overlay
+                    y: webView.privateMode ? _fullHeight : 0
+                    height: webView.fullscreenHeight
```
Comment (xlise) [inline apps/shared/OverlayAnimator.qml:176]: May be write values in other states

File: apps/browser/qml/pages/components/Overlay.qml [hunk 2]
```diff
@@ -321,7 +328,11 @@ Shared.Background {
                 textTopMargin: height/2 - _editor.implicitHeight/2
                 labelVisible: false
                 inputMethodHints: Qt.ImhUrlCharactersOnly
-                background: null
+                background: Rectangle{
```
Comment (xlise) [inline apps/browser/qml/pages/components/Overlay.qml:331]:
```suggestion
                background: Rectangle {
```

File: apps/shared/OverlayAnimator.qml [hunk 3]
```diff
@@ -190,15 +211,15 @@ Item {
...
+            to: "fullscreenWebPage,chromeVisible,loadProgressOverlay,fullscreenOverlay,noOverlay,doubleToolBar,certOverlay,startPage"
 
             SequentialAnimation {
                 NumberAnimation { target: webView; property: "height"; duration: transitionDuration; easing.type: Easing.InOutQuad }
                 ScriptAction {
                     script: {
                         if (animator.state === _chromeVisible || animator.state === _fullscreenWebPage || animator.state === _doubleToolBar) {
                             atBottom = true
-                        } else if (animator.state === _fullscreenOverlay || animator.state === _certOverlay) {
+                        } else if (animator.state === _fullscreenOverlay || animator.state === _certOverlay ||animator.state === _startPage ) {
```
Comment (xlise) [inline apps/shared/OverlayAnimator.qml:222]:
```suggestion
                        } else if (animator.state === _fullscreenOverlay || animator.state === _certOverlay || animator.state === _startPage ) {
```

**Review by llewelld — COMMENTED** [review]
This looks and works nicely generally. I have a few minor style comments.

One issue in relation to behaviour is that, on opening the browser when there are no existing tabs, the start page opens full screen. Shortly after the homepage loads and the start page disappears. Is this something that's been fixed elsewhere already? It looks like it's related to [this code](https://github.com/sailfishos/sailfish-browser/blob/master/apps/core/declarativewebcontainer.cpp#L902).

**Comment by aroditelev** [comment]
> This looks and works nicely generally. I have a few minor style comments.
> 
> One issue in relation to behaviour is that, on opening the browser when there are no existing tabs, the start page opens full screen. Shortly after the homepage loads and the start page disappears. Is this something that's been fixed elsewhere already? It looks like it's related to [this code](https://github.com/sailfishos/sailfish-browser/blob/master/apps/core/declarativewebcontainer.cpp#L902).

Если в настройках оставить пустое поле домашней страницы, то будет:
void DeclarativeWebContainer::initialize() homePage... "about:blank"
Если задать домашнюю страницу, то :

File: apps/browser/qml/pages/BrowserPage.qml [hunk 4]
```diff
@@ -294,7 +294,11 @@ Page {
         onActiveChanged: {
             var isFullScreen = webView.contentItem && webView.contentItem.fullscreen
             if (!isFullScreen && active && !overlay.enteringNewTabUrl) {
-                overlay.animator.showChrome()
+                if (webView.tabModel.count !== 0 || (WebUtils.homePage !== "about:blank" && WebUtils.homePage.length > 0)) {
```
Comment (aroditelev) [inline apps/browser/qml/pages/BrowserPage.qml:297]: @llewelld 
> This looks and works nicely generally. I have a few minor style comments.
> 
> One issue in relation to behaviour is that, on opening the browser when there are no existing tabs, the start page opens full screen. Shortly after the homepage loads and the start page disappears. Is this something that's been fixed elsewhere already? It looks like it's related to [this code](https://github.com/sailfishos/sailfish-browser/blob/master/apps/core/declarativewebcontainer.cpp#L902).

This change resolves the issue when using the home page.

Comment (llewelld) [reply apps/browser/qml/pages/BrowserPage.qml:297]: Great, this is much better. Thank you.

**Review by llewelld — APPROVED** [review]
This looks good to me. There is one further point I note, which is that it would be better if the start page opened fully after the user selects "Open new tab" (especially if there are no open tabs, since in this case it can't be dragged downwards anyway). Would this be easy to implemented?

If this requires notable new work, then I suggest we merge this as it is.

**Comment by aroditelev** [comment]
> This looks good to me. There is one further point I note, which is that it would be better if the start page opened fully after the user selects "Open new tab" (especially if there are no open tabs, since in this case it can't be dragged downwards anyway). Would this be easy to implemented?
> 
> If this requires notable new work, then I suggest we merge this as it is.

@llewelld  This will be relevant only if there is an active tab, if there are no tabs, then overlay will be opened in full screen.

**Comment by llewelld** [comment]
> @llewelld This will be relevant only if there is an active tab, if there are no tabs, then overlay will be opened in full screen.

This works correctly if I swipe back from the tab page, but not if I choose the "Open new tab" option. So, following these steps:
1. Open browser.
2. Press tab button on toolbar.
3. Select "Close all tabs" from context menu (wait for remorse timer to complete).
4. Select "Open new tab" from the context menu.

Now the overlay is not opened full screen.

File: apps/shared/OverlayAnimator.qml (hunk 3 above)
Comment (aroditelev) [reply apps/shared/OverlayAnimator.qml]: Done

**Comment by aroditelev** [comment]
> > @llewelld This will be relevant only if there is an active tab, if there are no tabs, then overlay will be opened in full screen.
> 
> This works correctly if I swipe back from the tab page, but not if I choose the "Open new tab" option. So, following these steps:
> 
> 1. Open browser.
> 2. Press tab button on toolbar.
> 3. Select "Close all tabs" from context menu (wait for remorse timer to complete).
> 4. Select "Open new tab" from the context menu.
> 
> Now the overlay is not opened full screen.

@llewelld Thank you, I really did not take into account such a case. Added a condition, now the home page opens when there are no active tabs.

**Comment by llewelld** [comment]
Reaffirming that these changes look good to me. Could you please rebase on next.

**Pull request merged**